src/components/KBadge/KBadge.vue

```diff
@@ -161,401 +99,305 @@ const props = defineProps({
-  $KBadgeDefaultColor: var(--kui-color-text-primary, $kui-color-text-primary);
-  $KBadgeDefaultButtonHoverColor: var(--kui-color-background-primary-weaker, $kui-color-background-primary-weaker);
-  // success appearance colors local variables
-  $KBadgeSuccessBackground: $tmp-color-green-100;
-  $KBadgeSuccessColor: $tmp-color-green-700;
-  $KBadgeSuccessButtonHoverColor: $tmp-color-green-200;
-  // danger appearance colors local variables
-  $KBadgeDangerBackground: var(--kui-color-background-danger-weakest, $kui-color-background-danger-weakest);
-  $KBadgeDangerColor: var(--kui-color-text-danger, $kui-color-text-danger);
-  $KBadgeDangerButtonHoverColor: var(--kui-color-background-danger-weaker, $kui-color-background-danger-weaker);
-  // info appearance colors local variables
-  $KBadgeInfoBackground: var(--kui-color-background-primary-weaker, $kui-color-background-primary-weaker);
-  $KBadgeInfoColor: var(--kui-color-text-primary, $kui-color-text-primary);
-  $KBadgeInfoButtonHoverColor: var(--kui-color-background-primary-weak, $kui-color-background-primary-weak);
-  // warning appearance colors local variables
-  $KBadgeWarningBackground: $tmp-color-yellow-100;
-  $KBadgeWarningColor: $tmp-color-yellow-600;
-  $KBadgeWarningButtonHoverColor: $tmp-color-yellow-200;
-  // neutral appearance colors local variables
-  $KBadgeNeutralBackground: var(--kui-color-background-neutral-weaker, $kui-color-background-neutral-weaker);
-  $KBadgeNeutralColor: var(--kui-color-text-neutral, $kui-color-text-neutral);
-  $KBadgeNeutralButtonHoverColor: var(--kui-color-background-neutral-weak, $kui-color-background-neutral-weak);
-
-   &.k-badge-custom {
-    background-color: v-bind('$props.backgroundColor');
-    border-color: v-bind('$props.borderColor');
-    color: v-bind('$props.color');
-
-    &.is-bordered {
-      border-style: solid;
-      border-width: var(--kui-border-width-10, $kui-border-width-10);
-    }
+  &.success {
+    background-color: var(--kui-color-background-success-weakest, $kui-color-background-success-weakest);
+    color: var(--kui-color-text-success, $kui-color-text-success);
 
-    .k-badge-dismiss-button {
-      background-color: v-bind('$props.backgroundColor');
-      .kong-icon.kong-icon-close path {
-        stroke: v-bind('$props.color');
+    :deep([role="button"]):not([disabled]) {
+      &:hover, &:focus {
+        color: var(--kui-color-text-success-strong, $kui-color-text-success-strong) !important;
       }
+    }
+  }
 
-      &:hover {
-        background-color: v-bind('$props.hoverColor');
+  &.warning {
+    background-color: var(--kui-color-background-warning-weakest, $kui-color-background-warning-weakest);
+    color: var(--kui-color-text-warning, $kui-color-text-warning);
+
+    :deep([role="button"]):not([disabled]) {
+      &:hover, &:focus {
+        color: var(--kui-color-text-warning-strong, $kui-color-text-warning-strong) !important;
       }
     }
+  }
```

Comment:
suggestion: since each appearance block appears to use the exact same CSS rule structure, would it be more efficient to add a mixin with params to compute each appearance so the code doesn't have to be replicated in each block?